src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.cc

```diff
@@ -687,18 +696,32 @@ StatusOr<struct openssl_symaddrs_t> OpenSSLSymAddrs(RawFptrManager* fptrManager,
   PL_ASSIGN_OR_RETURN(uint32_t openssl_fix_sub_version,
                       OpenSSLFixSubversionNum(fptrManager, openssl_lib, pid));
 
-  switch (openssl_fix_sub_version) {
-    case 0:
-      symaddrs.RBIO_num_offset = kOpenSSL_1_1_0_RBIO_num_offset;
-      break;
-    case 1:
-      symaddrs.RBIO_num_offset = kOpenSSL_1_1_1_RBIO_num_offset;
-      break;
-    default:
-      // Supported versions are checked in function OpenSSLFixSubversionNum(),
-      // should not fall through to here, ever.
-      DCHECK(false);
-      return error::Internal("Unsupported openssl_fix_sub_version: $0", openssl_fix_sub_version);
+  if (!IsBoringSSL(openssl_lib)) {
+    switch (openssl_fix_sub_version) {
+      case 0:
+        symaddrs.RBIO_num_offset = kOpenSSL_1_1_0_RBIO_num_offset;
+        break;
+      case 1:
+        symaddrs.RBIO_num_offset = kOpenSSL_1_1_1_RBIO_num_offset;
+        break;
+      default:
+        // Supported versions are checked in function OpenSSLFixSubversionNum(),
+        // should not fall through to here, ever.
+        DCHECK(false);
+        return error::Internal("Unsupported openssl_fix_sub_version: $0", openssl_fix_sub_version);
+    }
+  } else {
+    switch (openssl_fix_sub_version) {
+      case 1:
+        symaddrs.RBIO_num_offset = kBoringSSL_1_1_1_RBIO_num_offset;
+        symaddrs.SSL_rbio_offset = kBoringSSL_RBIO_offset;
+        break;
+      default:
+        // Supported versions are checked in function OpenSSLFixSubversionNum(),
+        // should not fall through to here, ever.
+        DCHECK(false);
+        return error::Internal("Unsupported openssl_fix_sub_version: $0", openssl_fix_sub_version);
+    }
```

Comment:
My hope is that if boringssl is linked and includes the `Openssl_version_num` symbol that we will be able to correctly determine the offsets without needing to rely on inspecting netty's (or another service's) version. I know this doesn't apply to envoy's case, but I thought this was worth a try first before building out the netty version detection.